wled00/json.cpp

```diff
@@ -292,23 +292,25 @@ bool deserializeState(JsonObject root, byte callMode, byte presetId)
 
   doReboot = root[F("rb")] | doReboot;
 
+  strip.setMainSegmentId(root[F("mainseg")] | strip.getMainSegmentId()); // must be before realtimeLock() if "live"
+
   realtimeOverride = root[F("lor")] | realtimeOverride;
   if (realtimeOverride > 2) realtimeOverride = REALTIME_OVERRIDE_ALWAYS;
 
-  bool liveEnabled = false;
   if (root.containsKey("live")) {
-    bool lv = root["live"];
-    if (lv) {
+    if (root["live"].as<bool>()) {
       transitionDelayTemp = 0;
       jsonTransitionOnce = true;
-      liveEnabled = true; // triggers realtimeLock() below
       realtimeLock(65000);
+    } else {
```

Comment:
I think I like this in principle, but I think there should be more ways to exit _indefinite_ live mode, e.g. turning off and on master power or setting an effect on mainseg if in mainseg only mode. Will think about it a little.